### PR TITLE
Automated cherry pick of #117238: api: encode NamespacedName with lower case in JSON

### PR DESCRIPTION
**Pull request by pohly**
Cherry pick of #117238 on release-1.27.

#117238: api: encode NamespacedName with lower case in JSON

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Comment by hakman** [comment]
@pohly @liggitt Could we get the approval to merge this?

**Comment by pohly** [comment]
The release managers (on CC) now need to approve the backport.

/cc @kubernetes/release-managers

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**
